go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -118,6 +124,91 @@ func TestComputeFilesWithXattr(t *testing.T) {
 	}
 }
 
+func TestComputeFileDigestWithXattr(t *testing.T) {
+	xattrDgName := "user.myhash"
+	overwriteXattrDgName(t, xattrDgName)
+	tests := []struct {
+		filename   string
+		contents   string
+		xattrDgStr string
+		wantDgStr  string
+		wantErr    error
+	}{
+		{
+			filename:  "provide_no_xattr_will_generate_real_digest(sha256sum+real_size)",
+			contents:  "123456",
+			wantDgStr: "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92/6",
+		},
+		{
+			filename:   "provide_only_digest_hash_will_generate_digest_with_real_size",
+			contents:   "123456",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/6",
+		},
+		{
+			filename:   "valid_full_digest_(hash+size)_in_xatrr_will_be_used_directly",
+			contents:   "",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111/666",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/666",
+		},
+		{
+			filename:   "provide_invalid_digest_(hash_missing_digits)_will_set_md.Err",
+			contents:   "123456",
+			xattrDgStr: "abc",
+			wantDgStr:  "abc/6",
+			wantErr:    &FileError{Err: errors.New("valid hash length is 64, got length 3 (abc)")},
```

Comment:
Matching exact error messages will make the test case fragile. It looks like in these cases just checking for an error is sufficient.
```suggestion
			wantErr:    true,
```
And then later:
`if !test.wantErr && err != nil { t.Fatalf("Unexpected error: %v", err) }`

Reply:
Thank you. My original plan is to use the error message to deliver some extra information to the users who are reading the compute() method and help them to understand the logic. Yeah, you are right, the error msg string will make the test fragile and depend on the implementation of digest package. I have change to this to bool now.